checker.c

```diff
@@ -1,18 +1,48 @@
 #include <stdio.h>
 #include <assert.h>
+#include "batteryManagement.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
+int checkChargeRate(float chargeRate)
+{
+   if(chargeRate > 0.8) {
+    printf("Charge Rate out of range!\n");
+    return 0;
+  }
+  else{
+    return 1;
+  }
+}
+int checkTemperature(float temperature)
+{
   if(temperature < 0 || temperature > 45) {
     printf("Temperature out of range!\n");
     return 0;
-  } else if(soc < 20 || soc > 80) {
+  } 
+  else{
+    return 1;
+  }
+}
+int checkSoc(float soc)
+{
+  if(soc < 20 || soc > 80) {
     printf("State of Charge out of range!\n");
     return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
+  } 
+  else{
+    return 1;
+  }
+}
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+   int batteryStatus=1;
+   batteryStatus &= checkChargeRate(chargeRate);
+   batteryStatus &= checkTemperature(temperature);
+   batteryStatus &= checkSoc(soc);
+  if(batteryStatus){
+      return 1;
+  }
+  else{
     return 0;
   }
-  return 1;
 }
 
 int main() {
```

Comment:
Try to cover boundary conditions, out of range values and in range values, for example test values for temperature could be -1, 0, 25, 45, 50. Same can be applied to other parameters too.